scipio/src/channels/local_channel.rs

```diff
@@ -207,6 +208,10 @@ struct SendWaiter<T> {
     channel: LocalChannel<T>,
 }
 
+struct RecvWaiter<T> {
+    channel: LocalChannel<T>,
+}
+
```

Comment:
These waiter structs look a bit silly to me. I wonder if we should leverage `poll_fn` here?

It works for SendWaiter, but doesn't for RecvWaiter, becuse it is used twice:

https://github.com/matklad/scipio/compare/4e21fddab2303ab58ba35c3f8bd873f6aefeedc4..a97e2b498dfef67e78dd6d938d6731746c33758d

Reply:
I've seen `poll_fn` before but wasn't terribly familiar with it.
I have read its docs again as a refresher and indeed seems like it could be used here.

I'll take another pass with this.

Thanks!